File: client/sdk/python/openai/chat_completions.py
"""OpenAI provider functions."""

from openai import OpenAI


def create_client(api_key, base_url):
    return OpenAI(api_key=api_key, base_url=base_url+"/v1")


def chat(prompt, model, api_key, base_url):
    client = create_client(api_key=api_key, base_url=base_url)
    response = client.chat.completions.create(
        model=model,
        messages=[{"role": "user", "content": prompt}],
    )
    return response.choices[0].message.content


def stream_chat(prompt, model, api_key, base_url):
    client = create_client(api_key=api_key, base_url=base_url)
    stream = client.chat.completions.create(
        model=model,
        messages=[{"role": "user", "content": prompt}],
        stream=True,
    )
    for chunk in stream:
        if chunk.choices and len(chunk.choices) > 0:
            delta = chunk.choices[0].delta.content
            if delta:
                yield delta

File: client/sdk/typescript/src/openai/chat_completions.ts
import type { ClientConfig } from "../config";

import { createClient } from "./shared";

export async function chat(config: ClientConfig, prompt: string, model: string): Promise<string> {
  const client = createClient(config);
  const response = await client.chat.completions.create({
    model,
    messages: [{ role: "user", content: prompt }],
  });
  return response.choices[0]?.message?.content ?? "";
}

export async function* streamChat(
  config: ClientConfig,
  prompt: string,
  model: string,
): AsyncIterable<string> {
  const client = createClient(config);
  const stream = await client.chat.completions.create({
    model,
    messages: [{ role: "user", content: prompt }],
    stream: true,
  });

  for await (const chunk of stream) {
    const delta = chunk.choices[0]?.delta?.content;
    if (delta) {
      yield delta;
    }
  }
}
